debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresSnapshotChangeEventSource.java

```diff
@@ -293,6 +293,12 @@ protected void setSnapshotTransactionIsolationLevel(boolean isOnDemand) throws S
         } else {
             LOGGER.info("Skipping setting snapshot time, snapshot data will not be consistent");
         }
+
+        // Regardless of whether consistent snapshot is enabled or not, we need to set the
+        // transaction isolation level.
+        String transactionIsolationLevelStatement = "SET SESSION CHARACTERISTICS AS TRANSACTION ISOLATION LEVEL SERIALIZABLE, READ ONLY, DEFERRABLE;";
```

Comment:
Why do we set this across session? If consistent snapshot is enabled we run the following `SET TRANSACTION ISOLATION LEVEL SERIALIZABLE, READ ONLY, DEFERRABLE;` Will this not work ?

Reply:
The setting of session characteristics is from an internal discussion and that's the reason I proceeded with that.